src/rime/config/config_compiler.cc

```diff
@@ -127,13 +127,21 @@ bool PendingChild::Resolve(ConfigCompiler* compiler) {
 static an<ConfigItem> ResolveReference(ConfigCompiler* compiler,
                                        const Reference& reference);
 
+static bool MergeTree(an<ConfigItemRef> target, an<ConfigMap> map);
+
 bool IncludeReference::Resolve(ConfigCompiler* compiler) {
   DLOG(INFO) << "IncludeReference::Resolve(reference = " << reference << ")";
-  auto item = ResolveReference(compiler, reference);
-  if (!item) {
+  auto included = ResolveReference(compiler, reference);
+  if (!included) {
     return reference.optional;
   }
-  *target = item;
+  // merge literal key-values into the included map
+  auto overrides = As<ConfigMap>(**target);
```

Comment:
Can this not be a map?
A list?
A scalar value?

It seems in YAML the neighbor of `__include: *` has to be other key-values in the same map.
But the included node can be any type: null, scalar, list or map.

Reply:
list appended to included list must be introduced by `__append`. because yaml syntax doesn't allow list items there.
```yaml
list:
  - 1
  - 2

new_list:
  __include: list
  __append:
    - 3
    - 4
```

similarly `MergeTree` can produce a scalar by string concatenation.

```yaml
path: '/bin:/usr/bin'

more_path:
   __include: path
   __append: ':/usr/local/bin'
```

in these cases, it's an error to provide keys other than `__include` and `__append` in the map